### PR TITLE
New Version endpoint with support for the data-mover 'version' and a …

### LINKED ISSUE #28: Copy Offload - Client / Server API Compatibility Check
The API to the data movement service is being exported to a c-library that may be linked in with other using executables.  It would be helpful if the c-library could determine whether or not it is compatible with the service it is attempting to communicating with.

For example, if the client was originally created and compiled for a specific version of the server, and the current server is running an incompatible version, it would be best if the client could return an error code to the using program indicating the incompatibility.

Without this check, it is possible for the incompatible client to attempt to communicate with the newer server and produce unexpected behavior.

I wonder if this check could be implemented within the Connect implementation and returned as an error code there.

I had originally wrote this as an email with a specific request for a version API.  However, after giving it more thought, it is really the compatibility check and error return that I am looking for as it may help with deployment concerns.

### PR DESCRIPTION
**Pull request by NateThornton**
New Version endpoint with support for the data-mover 'version' and a list of supported 'API versions'
Example client snippet

```c++
    {
        // Retrieve the version information

        VersionResponse versionResponse;

        RPCStatus status = client.Version(&versionResponse);
        if (!status.ok()) {
            std::cout << "Version RPC FAILED (" << status.error_code() << "): " << status.error_message() << std::endl;
            return 1;
        }

        std::cout << "Data Movement Version: " << versionResponse.version() << std::endl;
        std::cout << "Supported API Versions:" << std::endl;
        for (auto version : versionResponse.apiversions()) {
            std::cout << "\t" << version << std::endl;
        }

    }
```

```
$ ./client 
Data Movement Version: v0.0.1-alpha
Supported API Versions:
        1
```

Closes #28

Signed-off-by: Nate Thornton <nate.thornton@hpe.com>

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -16,4 +16,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: ghcr.io/nearnodeflash/nnf-dm
-  newTag: 0.0.1
+  newTag: 2cda27e6cf229a7e5875790463794a81367fcdcc
```
Comment (bdevcich-hpe) [inline config/manager/kustomization.yaml:19]: Don't want this.

File: daemons/compute/lib-cpp/example/client.cc [hunk 2]
```diff
@@ -113,7 +132,7 @@ int main(int argc, char** argv) {
 
         RPCStatus status = client.Cancel(workflow, cancelRequest, &cancelResponse);
         if (!status.ok()) {
-            std::cout << "Cancle RPC FAILED" << status.error_code() << ": " << status.error_message() << std::endl;
+            std::cout << "Cancle RPC FAILED (" << status.error_code() << "): " << status.error_message() << std::endl;
```
Comment (bdevcich-hpe) [inline daemons/compute/lib-cpp/example/client.cc:135]:
```suggestion
            std::cout << "Cancel RPC FAILED (" << status.error_code() << "): " << status.error_message() << std::endl;
```

File: daemons/compute/server/version/version.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package version
+
+// The current data-mover version string. Clients can use this to identify particular
+// information about the installed data-mover daemon.
+var (
+	// Version contains the current version of the data mover daemon.
+	version = "v0.0.1"
```
Comment (bdevcich-hpe) [inline daemons/compute/server/version/version.go:26]: Can this eventually get pulled from kustomization.yaml (or somewhere)? Perhaps the same thing with the commitHash below?

**Review by bdevcich-hpe — APPROVED** [review]

**Review by matthew-richerson — APPROVED** [review]

File: daemons/compute/server/version/version.go (hunk 3 above)
Comment (NateThornton) [reply daemons/compute/server/version/version.go:26]: This is the version of the data-mover daemon; kubernetes is not involved. Are you asking for a kuberntes data-movement resource value?

`commitHash` comes from the git commit. I'm not sure what you're asking here.

**Pull request merged**